### PR TITLE
Microsoft.ProviderHub: Add new API version 2021-06-01-preview 

### PR DESCRIPTION
**Pull request by diwudd**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [X] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by sanjaiganesh** [comment]
>     "/subscriptions/{subscriptionId}/providers/Microsoft.ProviderHub/providerRegistrations/{providerNamespace}/resourceActions/{resourceActionName}/getResource": {

/listResources ?

---
Refers to: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/providerhub.json:2139 in c079edd. [](commit_id = c079edd8ea8031350fff9b2e3e2899559807bf37, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>     "/subscriptions/{subscriptionId}/providers/Microsoft.ProviderHub/providerRegistrations/{providerNamespace}/resourceActions/{resourceActionName}/getResource": {

/resourceOperations/default/listResources ?

---
In reply to: [854133036](https://github.com/Azure/azure-rest-api-specs/pull/14650#issuecomment-854133036) [](http://example.com/codeflow?ancestors=854133036)

---
Refers to: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/providerhub.json:2139 in 925a247. [](commit_id = 925a2476a1608f2ed7de422111492e6bdfc39576, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>     "/subscriptions/{subscriptionId}/providers/Microsoft.ProviderHub/providerRegistrations/{providerNamespace}/resourceActions/{resourceActionName}/getResources": {

/resourceOperations/default/listResources?

---
Refers to: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/providerhub.json:2194 in c079edd. [](commit_id = c079edd8ea8031350fff9b2e3e2899559807bf37, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>     "/subscriptions/{subscriptionId}/providers/Microsoft.ProviderHub/providerRegistrations/{providerNamespace}/resourceActions/{resourceActionName}/getResources": {

remove {resourceActionName} parameter

---
In reply to: [854261751](https://github.com/Azure/azure-rest-api-specs/pull/14650#issuecomment-854261751) [](http://example.com/codeflow?ancestors=854261751)

---
Refers to: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/providerhub.json:2194 in 925a247. [](commit_id = 925a2476a1608f2ed7de422111492e6bdfc39576, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>               "$ref": "#/definitions/ResourceManagementEntity"

Take a different payload for list call, including OData filter that current LIST call for cross partition query takes.

---
Refers to: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/providerhub.json:2219 in c079edd. [](commit_id = c079edd8ea8031350fff9b2e3e2899559807bf37, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>               "$ref": "#/definitions/ResourceManagementEntity"

check ARG list call that is similar to this.
 $ref": "#/definitions/QueryRequest"

POST providers/Microsoft.ResourceGraph/resources

---
In reply to: [854262122](https://github.com/Azure/azure-rest-api-specs/pull/14650#issuecomment-854262122) [](http://example.com/codeflow?ancestors=854262122)

---
Refers to: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/providerhub.json:2219 in 925a247. [](commit_id = 925a2476a1608f2ed7de422111492e6bdfc39576, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>     "/subscriptions/{subscriptionId}/providers/Microsoft.ProviderHub/providerRegistrations/{providerNamespace}/resourceActions/{resourceActionName}/getResources": {

consdier moving under /ResourceTypeRegistrations/ as John  suggested, as it will allow ARM to RBAC the user based on the Role. This is useful with multi-registration model

---
Refers to: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/providerhub.json:2194 in c079edd. [](commit_id = c079edd8ea8031350fff9b2e3e2899559807bf37, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>           }

I suppose cascade deletion implicit here.

---
Refers to: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/providerhub.json:5000 in c079edd. [](commit_id = c079edd8ea8031350fff9b2e3e2899559807bf37, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>               "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"

it could be tracked resource as well. 

---
Refers to: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/providerhub.json:2176 in c079edd. [](commit_id = c079edd8ea8031350fff9b2e3e2899559807bf37, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>               "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"

use TrackedResource definition, unless it has location as required property in the v2 definition

---
In reply to: [854264300](https://github.com/Azure/azure-rest-api-specs/pull/14650#issuecomment-854264300) [](http://example.com/codeflow?ancestors=854264300)

---
Refers to: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/providerhub.json:2176 in 925a247. [](commit_id = 925a2476a1608f2ed7de422111492e6bdfc39576, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>         "x-ms-long-running-operation": false

nit - remove. it is not long running. 

---
Refers to: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/providerhub.json:2191 in c079edd. [](commit_id = c079edd8ea8031350fff9b2e3e2899559807bf37, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>         "x-ms-long-running-operation": false

nit - remove

---
Refers to: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/providerhub.json:2246 in c079edd. [](commit_id = c079edd8ea8031350fff9b2e3e2899559807bf37, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>             }

no response body for 202 

---
Refers to: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/providerhub.json:2287 in c079edd. [](commit_id = c079edd8ea8031350fff9b2e3e2899559807bf37, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>           "202": {

is it always async ?

---
Refers to: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/providerhub.json:2283 in c079edd. [](commit_id = c079edd8ea8031350fff9b2e3e2899559807bf37, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>               "$ref": "#/definitions/ResourceManagementAction"

should this take a an option to invoke UserRP extensions ? 
Since UserRP may have buisiness logic that they would like to invoke

---
Refers to: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/providerhub.json:2274 in c079edd. [](commit_id = c079edd8ea8031350fff9b2e3e2899559807bf37, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>             "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"

same here. it may not always be proxy

---
Refers to: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/providerhub.json:5010 in c079edd. [](commit_id = c079edd8ea8031350fff9b2e3e2899559807bf37, deletion_comment = False)

**Comment by sanjaiganesh** [comment]
>         }

would the caller do a GET to follow the nextLink ? What would it look like 


---
Refers to: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/providerhub.json:5016 in c079edd. [](commit_id = c079edd8ea8031350fff9b2e3e2899559807bf37, deletion_comment = False)

**Comment by weidongxu-microsoft** [comment]
What is the change of the PR (exclude mere api-version update)?

**Comment by diwudd** [comment]
It also contains a new api. You can find it by comparing the latest iteration with the first iteration.

Thanks,

Di

File: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/examples/CheckinManifest.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "parameters": {
+    "subscriptionId": "ab7a8701-f7ef-471a-a2f4-d0ebbf494f77",
+    "providerNamespace": "Microsoft.Contoso",
+    "api-version": "2021-05-01-preview",
```
Comment (RamyasreeChakka) [inline specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/examples/CheckinManifest.json:5]: @diwudd can you update API version in all the example files?

**Comment by diwudd** [comment]
Sure, updated all of them.

Thanks,

Di

**Comment by diwudd** [comment]
Hi,

Can you please take another look at this PR? The breaking change has been approved.

Thanks,

Di

From: Di Wu
Sent: Tuesday, June 22, 2021 10:38 AM
To: Azure/azure-rest-api-specs ***@***.***>; Azure/azure-rest-api-specs ***@***.***>
Cc: Mention ***@***.***>
Subject: RE: [Azure/azure-rest-api-specs] Microsoft.ProviderHub: Add new API version 2021-06-01-preview (#14650)

Sure, updated all of them.

Thanks,

Di

**Comment by weidongxu-microsoft** [comment]
You need to pass ARM review.

**Comment by diwudd** [comment]
Hi Weidong,

I saw you are the assigned to review this PR. Will you review it or can you please add ARM folks to review it?

Thanks,

Di

File: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/examples/CheckinManifest.json (hunk 1 above)
Comment (diwudd) [reply specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/examples/CheckinManifest.json]: Updated. Can you please take another look?

**Comment by RamyasreeChakka** [comment]
@diwudd - Changed LGTM. Signed off from ARM side.

File: specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/examples/CheckinManifest.json (hunk 1 above)
Comment (RamyasreeChakka) [reply specification/providerhub/resource-manager/Microsoft.ProviderHub/preview/2021-06-01-preview/examples/CheckinManifest.json]: @diwudd - signed off from ARM side.

**Comment by diwudd** [comment]
Thank you!

Can you help merge it as I don't have the permissions? Or will someone else merge it?

Thanks,

Di

**Comment by RamyasreeChakka** [comment]
> @diwudd Di Wu FTE - Changed LGTM. Signed off from ARM side.

@diwudd - Please work with your PR assignee for merge. I don't have permissions.

**Comment by diwudd** [comment]
Hi Weidong,

The PR had been signed off by ARM team.

Can you please merge it?

Thanks,

Di

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**